tests/unit/test_parameters/test_process_parameter_data.py

```diff
@@ -2,56 +2,59 @@
 # Tests for the parameter processing functions
 #
 
-
 import os
 import numpy as np
 import pybamm
-
 import pytest
 
 
+@pytest.fixture
+def path():
+    return os.path.abspath(os.path.dirname(__file__))
+
+
+@pytest.fixture(
+    params=[
+        ("lico2_ocv_example", pybamm.parameters.process_1D_data),
+        ("lico2_diffusivity_Dualfoil1998_2D", pybamm.parameters.process_2D_data),
+        ("data_for_testing_2D", pybamm.parameters.process_2D_data_csv),
+        ("data_for_testing_3D", pybamm.parameters.process_3D_data_csv),
+    ]
+)
+def parameter_data(request, path):
+    name, processing_function = request.param
+    processed = processing_function(name, path)
+    return name, processed
+
+
 class TestProcessParameterData:
-    def test_process_1D_data(self):
-        name = "lico2_ocv_example"
-        path = os.path.abspath(os.path.dirname(__file__))
-        processed = pybamm.parameters.process_1D_data(name, path)
+    def test_processed_name(self, parameter_data):
+        name, processed = parameter_data
         assert processed[0] == name
-        assert isinstance(processed[1], tuple)
-        assert isinstance(processed[1][0][0], np.ndarray)
-        assert isinstance(processed[1][1], np.ndarray)
 
-    def test_process_2D_data(self):
-        name = "lico2_diffusivity_Dualfoil1998_2D"
-        path = os.path.abspath(os.path.dirname(__file__))
-        processed = pybamm.parameters.process_2D_data(name, path)
-        assert processed[0] == name
-        assert isinstance(processed[1], tuple)
-        assert isinstance(processed[1][0][0], np.ndarray)
-        assert isinstance(processed[1][0][1], np.ndarray)
-        assert isinstance(processed[1][1], np.ndarray)
+    def test_processed_structure(self, parameter_data):
+        """
+        Test that the processed data has the correct structure.
```

Comment:
```suggestion
```

Reply:
i.e., we generally avoid documenting a test method because most of them are usually quite readable and test a small thing across the codebase (hence the name – "unit" tests).